packages/edit-site/src/components/global-styles/screen-link-color.js

```diff
@@ -29,40 +34,87 @@ function ScreenLinkColor( { name } ) {
 		isLinkEnabled &&
 		( solids.length > 0 || areCustomSolidsEnabled );
 
-	const [ linkColor, setLinkColor ] = useStyle(
-		'elements.link.color.text',
-		name
-	);
-	const [ userLinkColor ] = useStyle(
-		'elements.link.color.text',
-		name,
-		'user'
-	);
+	const pseudoStates = {
+		default: {
+			...useStyle( 'elements.link.color.text', name ).reduce(
+				( acc, psuedoSelector, index ) => {
+					acc[ index === 0 ? 'value' : 'handler' ] = psuedoSelector;
+					return acc;
+				},
+				{}
+			),
+			userValue: useStyle(
+				'elements.link.color.text',
+				name,
+				'user'
+			)[ 0 ],
```

Comment:
This avoids the need for creating lots of intermediate variables. Annoyingly the hooks return in the `[X, Y]` form so we need to map to object keys and spread into the config.

I did this because in future we'll need to dynamically generate the psuedo states config from an array of psuedo classes (e.g. `['hover','focus','active']` .etc).

Reply:
I find this snippet with the spread reducer very confusing. I'd rather have it done before and assigned to variable names that help one understand what the reducer is after.

Reply:
Yeh that's what it used to do. I'm thinking perhaps it's an abstraction too far at this point. Unless I can find a more elegant way to do it.

Reply:
@draganescu I simplified it in https://github.com/WordPress/gutenberg/pull/41976/commits/89ab25e2d4e465aec002cc9b596a2046b8572fb2. Should be easy to follow now.